portfolio/src/test/java/FiltersTest.java

```diff
@@ -0,0 +1,69 @@
+import static org.junit.Assert.assertEquals;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.SearchQuery;
+import java.util.Arrays;
+import java.util.List;
+import java.net.MalformedURLException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+public class FiltersTest {
```

Comment:
Could we see some tests for the edge cases? Empty titles, empty descriptions, empty keywordLists, etc.

Reply:
Yeah I can do that. Just to provide some context, I discussed with Quinn and we haven't been able to run the Natural Language API with JUnit so I won't be able to test certain parts of the SearchQuery class.